### PR TITLE
prometheus-xmpp-alerts: 0.4.2 -> 0.5.1; apply RFC 42

### PR DESCRIPTION
**Pull request by hax404**
###### Motivation for this change
Fix 500s that come back when the alertmanager sends alerts

```
prometheus-xmpp-alerts[30888]: ERROR    Error handling request
prometheus-xmpp-alerts[30888]: Traceback (most recent call last):
prometheus-xmpp-alerts[30888]:   File "/nix/store/nlvjjvb9471dbpdj8fhdih0ak90cq9d9-python3.8-aiohttp-3.6.2/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 418, in start
prometheus-xmpp-alerts[30888]:     resp = await task
prometheus-xmpp-alerts[30888]:   File "/nix/store/nlvjjvb9471dbpdj8fhdih0ak90cq9d9-python3.8-aiohttp-3.6.2/lib/python3.8/site-packages/aiohttp/web_app.py", line 458, in _handle
prometheus-xmpp-alerts[30888]:     resp = await handler(request)
prometheus-xmpp-alerts[30888]:   File "/nix/store/n2axls35979gx2aqdfnl0yafnrffl100-prometheus-xmpp-alerts-0.4.2/bin/.prometheus-xmpp-alerts-wrapped", line 141, in serve_alert
prometheus-xmpp-alerts[30888]:     text = '\n--\n'.join(create_message_full(alert))
prometheus-xmpp-alerts[30888]:   File "/nix/store/n2axls35979gx2aqdfnl0yafnrffl100-prometheus-xmpp-alerts-0.4.2/lib/python3.8/site-packages/prometheus_xmpp/__init__.py", line 57, in create_message_full
prometheus-xmpp-alerts[30888]:     alert['annotations']['summary'],
prometheus-xmpp-alerts[30888]: KeyError: 'summary'
prometheus-xmpp-alerts[30888]: INFO     ::1 [11/Oct/2020:20:18:38 +0000] "POST /alert HTTP/1.1" 500 244 "-" "Alertmanager/0.21.0"
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by maralorn** [comment]
Result of `nixpkgs-review pr 100274` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>prometheus-xmpp-alerts</li>
  </ul>
</details>

**Comment by maralorn** [comment]
The fix is apparently in this commit: https://github.com/jelmer/prometheus-xmpp-alerts/commit/8b4c2f7bd2c548a751748b913ce0f45b9f49c6ea but I see no harm in just bumping to the newest commit. It includes a commit by @andir ;-) I saw.

This looks fine by me. I am just not sure if there is some other convention for the version name that we should use.

I would like a second opinion on that last point, but ping me, if you don‘t get any more reactions.

File: pkgs/servers/monitoring/prometheus/xmpp-alerts.nix
Comment (mweinelt) [inline pkgs/servers/monitoring/prometheus/xmpp-alerts.nix:5]: From the traceback this looks alot like an upstream issue, so the proper course of action would be to report it upstream and have them make a new release.

In the meantime we could apply a patch to fix the situation if you can identify one.

**Review by mweinelt — CHANGES_REQUESTED** [review]
Please report this issue upstream and have them issue a fixed release.

**Comment by maralorn** [comment]
@mweinelt The issue is already fixed upstream. (As I pointed to in the commit I linked.)

Do you want to wait with this PR until they have made the next release?

**Comment by mweinelt** [comment]
I'm saying let's apply the patch and wait for upstream to do a new release. Let's not start doing releases downstream.

**Comment by maralorn** [comment]
Fair enough.

**Comment by andir** [comment]
You want at least 0f54da76ecd200887e7f267113b7c6a9cfae43d4 & 500a316a83b5d72c490cb62c3e7f5aa5efc7f5bf from my fork in order to actually be able to use the amtool integration of this. I filed a PR for those just now: https://github.com/jelmer/prometheus-xmpp-alerts/pull/19

File: pkgs/servers/monitoring/prometheus/xmpp-alerts.nix
Comment (andir) [inline pkgs/servers/monitoring/prometheus/xmpp-alerts.nix:14]: This is wrong. You only need pytz during the check phase. You are also missing the `aiohttp` runtime requirement.

**Review by andir — CHANGES_REQUESTED** [review]

**Comment by andir** [comment]
While we are waiting for upstream and/or the required fixes you can probably start adding a nixos module based on https://github.com/andir/infra/blob/e078bda5b9dafbfff2216a8e9131ce180196c362/config/servers/mon/xmpp-alerts.nix.

**Comment by andir** [comment]
Please update to the latest release: https://github.com/jelmer/prometheus-xmpp-alerts/commit/1620e9ea7bb2bfc66f033e35765a996473deabf3

**Comment by SuperSandro2000** [comment]
@hax404 ping

**Comment by maralorn** [comment]
LGTM, but other people in this thread are more familiar with these parts of nixpkgs.

File: pkgs/servers/monitoring/prometheus/xmpp-alerts.nix
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/xmpp-alerts.nix:9]:
```suggestion
buildPythonApplication rec {
```
applying with to the entire scope is an anti pattern.

File: pkgs/servers/monitoring/prometheus/xmpp-alerts.nix
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/xmpp-alerts.nix:23]:
```suggestion
  ] ++ (with python3Packages; [
```

File: nixos/modules/services/monitoring/prometheus/xmpp-alerts.nix
Comment (mweinelt) [inline nixos/modules/services/monitoring/prometheus/xmpp-alerts.nix:21]:
```suggestion
        for supported values.
```

nit :wink:

File: nixos/modules/services/monitoring/prometheus/xmpp-alerts.nix
Comment (mweinelt) [inline nixos/modules/services/monitoring/prometheus/xmpp-alerts.nix:14]:
```suggestion
    settings = mkOption {
```

and then add the following to add a migration path between old and new option.

```
  imports = [
    (mkRenamedOptionModule 
      [ "services" "prometheus" "xmpp-alerts" "configuration" ]
      [ "services" "prometheus" "xmpp-alerts" "settings" ])
  ];
```

File: nixos/modules/services/monitoring/prometheus/xmpp-alerts.nix
Comment (mweinelt) [inline nixos/modules/services/monitoring/prometheus/xmpp-alerts.nix:8]:
```suggestion
  configFile = settingsFormat.generate "prometheus-xmpp-alerts.yml" cfg.settings;
```

**Review by mweinelt — COMMENTED** [review]
The package looks fine, the module can be migrated further to RFC 42 still, also in a backward compatible way.

**Pull request merged**
